compiler/src/compiler/hir_to_mir.rs

```diff
@@ -611,6 +611,59 @@ impl<'a> PatternLoweringContext<'a> {
                     self.compile_match_conjunction(body, condition_builders);
                 })
             }
+            hir::Pattern::Or(patterns) => {
+                let [first_pattern, rest_patterns @ ..] = patterns.as_slice() else {
+                    panic!("Or pattern must contain at least two patterns.");
+                };
+
+                let mut result = self.compile(body, expression, first_pattern);
+
+                let captured_identifiers_order = first_pattern.captured_identifiers();
+                let list_get_function = body.push_builtin(BuiltinFunction::ListGet);
+                let nothing = body.push_nothing();
+                dbg!(&captured_identifiers_order);
+
+                for pattern in rest_patterns {
+                    let is_match = body.push_is_match(result, self.responsible);
+                    result = body.push_if_else(
+                        is_match,
+                        |body| {
+                            let captured_identifiers = pattern.captured_identifiers();
+                            dbg!(&captured_identifiers);
+                            if captured_identifiers == captured_identifiers_order {
+                                body.push_reference(result);
+                            } else {
+                                let captured_identifiers = captured_identifiers_order
+                                    .iter()
+                                    .map(|identifier_id| {
+                                        dbg!(identifier_id);
+                                        captured_identifiers
+                                            .iter()
+                                            .position(|it| it == identifier_id)
+                                            .map(|index| {
+                                                dbg!(index);
+                                                let index = body.push_int((1 + index).into());
+                                                body.push_symbol("Blub".to_string());
+                                                body.push_call(
+                                                    list_get_function,
+                                                    vec![result, index],
+                                                    self.responsible,
```

Comment:
Also, the code is very far indented here. If you know a way to fix it, sure, otherwise this is fine